src/test/components/CallTreeSidebar.test.js

```diff
@@ -179,12 +188,24 @@ describe('CallTreeSidebar', function () {
       queryByText,
       getAllByText,
       funcNamesDict: { A, B, C },
+      getState,
+      rerenderContainer,
     } = setup(getProfileWithSubCategories());
     selectNode([A, B, C]);
     expect(queryByText('FakeSubCategoryC')).not.toBeInTheDocument();
+    expect(container.firstChild).toMatchSnapshot();
+    expect(getState().urlState.profileSpecific.sidebarOpenCategories).toEqual(
+      new Set()
+    );
 
     const layoutCategory = getAllByText('Layout')[0];
+
     fireFullClick(layoutCategory);
+    rerenderContainer();
```

Comment:
I'm very surprised that we need to rerender here. This may be because of how you deal with the own props and the state props...

Reply:
It's not the only case in this project, but yes maybe something with the Props.

Reply:
this isn't the case